skill/wallpaper.py

```diff
@@ -0,0 +1,104 @@
+from datetime import datetime
+from pathlib import Path
+
+import requests
+
+DEFAULT_WALLPAPER = "blackwater-river.png"
+
+
+class Wallpaper:
+    """
+    Attributes:
+        collection: a list of paths to wallpaper files found on the device
+        default: the default wallpaper as defined in the skill settings
+        selected: the currently active (selected) wallpaper
+        skill_directory: the wallpaper directory in the home screen skill
+        skill_data_directory: a directory on the device for storing user-defined wallpapers
+    """
+
+    def __init__(self, skill_root_dir: str, skill_data_dir: str):
+        self.skill_directory = Path(skill_root_dir).joinpath("ui/wallpapers")
+        self.skill_data_directory = Path(skill_data_dir).joinpath("wallpapers")
+        self._ensure_user_directory_exists()
+        self.default = self._get_file_path(DEFAULT_WALLPAPER)
+        self.selected = self.default
+        self.collection = []
+        self.collect()
+
+    def _ensure_user_directory_exists(self):
+        """Ensures the directory for user-defined wallpapers exists."""
+        if not self.skill_data_directory.exists():
+            self.skill_data_directory.mkdir()
+
+    def collect(self):
+        """Builds a list of wallpapers provided by the skill and added by the user."""
+        for wallpaper_path in self.skill_directory.iterdir():
+            self.collection.append(wallpaper_path)
+        for wallpaper_path in self.skill_data_directory.iterdir():
+            self.collection.append(wallpaper_path)
+
+    def change(self, file_name: str):
+        """Change the wallpaper to use the image in a specified file.
+
+        Args:
+            file_name: the name of the wallpaper file
+        """
+        file_path = self._get_file_path(file_name)
+        if file_path in self.collection:
+            self.selected = file_path
+        else:
+            self.selected = self.default
+
+    def next(self):
+        """Selects the next wallpaper in the collection for display.
+
+        Starts over the beginning of the list after the last in the collection.  If
+        the "selected" attribute contains a value not in the collection, the default
+        is selected.
+        """
+        if self.selected in self.collection:
+            if self.selected == self.collection[-1]:
+                self.selected = self.collection[0]
+            else:
+                index_of_selected = self.collection.index(self.selected)
+                self.selected = self.collection[index_of_selected + 1]
+        else:
+            self.selected = self.default
+
+    def add(self, wallpaper_url: str):
+        """Adds a new wallpaper to the collection and selects it for display.
+
+        Args:
+            wallpaper_url: the url to download the wallpaper image from
+        """
+        response = requests.get(wallpaper_url)
+        file_path = self.skill_data_directory.joinpath("wallpaper.jpg")
```

Comment:
Just a suggestion - would `custom-wallpaper.jpg` or something be clearer if someone is looking at their filesystem?
#unimportantnitpick

Reply:
I can get down with that.  fixing.